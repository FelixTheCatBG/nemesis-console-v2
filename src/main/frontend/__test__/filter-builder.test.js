import FilterBuild from '../src/app/services/filter-builder';

describe('test filter build for null and not null restrictions', () => {
  test('should allowed empty value for "is null" filter', () => {
    let filterFields = [{field: 'some null field', restriction: 'Null', value: null}];
    expect(FilterBuild.buildFilter(filterFields)).toMatchSnapshot();
  });

  test('should allowed empty value for "is not null" filter', () => {
    let filterFields = [{field: 'some not null field', restriction: 'NotNull', value: null}];
    expect(FilterBuild.buildFilter(filterFields)).toMatchSnapshot();
  });

  test('should result with empty value and non empty for "is null" filter to be the same', () => {
    let fieldWithNullValue = {field: 'some null field', restriction: 'Null', value: null};
    let fieldWithValue = {field: 'some null field', restriction: 'Null', value: 'test'};

    expect(FilterBuild.buildFilter([fieldWithNullValue])).toBe(FilterBuild.buildFilter([fieldWithValue]));
  });

  test('should result with empty value and non empty for "is not null" filter to be the same', () => {
    let fieldWithNullValue = {field: 'some not null field', restriction: 'NotNull', value: null};
    let fieldWithValue = {field: 'some not null field', restriction: 'NotNull', value: 'test'};

    expect(FilterBuild.buildFilter([fieldWithNullValue])).toBe(FilterBuild.buildFilter([fieldWithValue]));
  });
});