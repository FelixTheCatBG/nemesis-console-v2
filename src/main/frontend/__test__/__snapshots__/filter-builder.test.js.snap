// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test filter build for null and not null restrictions should allowed empty value for "is not null" filter 1`] = `"some not null field ne null"`;

exports[`test filter build for null and not null restrictions should allowed empty value for "is null" filter 1`] = `"some null field eq null"`;
